MIPs/mip-x-authorize.md

```diff
@@ -0,0 +1,294 @@
+---
+mip: x
+title: Wallet Authorization for Multiple EVM Networks
+status: Draft
+stability: n/a
+discussions-to: https://github.com/MetaMask/metamask-improvement-proposals/discussions/33
+author(s): Alex Donesky (@adonesky1), Jiexi Luan (@jiexi), Shane Jonas (@shanejonas), Vandan Parikh (@vandan) 
+type: Maintainer
+created: 2024-01-25
+---
+
+## Summary
+
+A proposal to introduce an unrestricted `wallet_authorize` method for applications to connect to MetaMask per ChainAgnostic Standards Alliance’s CAIP-25 standard. The `wallet_authorize` method establishes an interface for authorizing application access to wallet-managed resources including accounts, RPC providers, and specific JSON-RPC methods across multiple EVM-compatible networks. An application can request access to these resources by specifying scopes as defined in CAIP-217.
+
+This method is intended to be part of a new version of the MetaMask Wallet API that is optimized for multinetwork interactions.
+
+This proposal aims to implement CAIP-25's JSON-RPC Wallet Authorization for EVM-compatible networks into MetaMask. For these networks, the `rpcEndpoints` parameter is adapted to be compliant with the object structure defined in EIP-3085. Designing the interface in this way will provide a standardized approach for applications to request access to networks and JSON-RPC methods.
+
+## Motivation
+
+The purpose of this proposal is to offer a convenient and standard pattern for applications to negotiate multiple network authorizations with MetaMask. 
+
+The `wallet_authorize` method is specifically meant to:
+
+- Allow applications to progressively negotiate a set of authorizations with MetaMask
```

Comment:
progressively implies being able to update existing authorizations. I believe that this method replaces all existing authorizations?

Reply:
This MIP is still awaiting some resolution and guidance from the lifecycle CAIP being proposed right now.

Reply:
I believe there are two approaches being considered:

1. Use the same `wallet_authorize` method for initial and progressive (incremental) authorizations
2. Use one method for the initial authorization (replaces any existing authorizations) and another method for progressive authorizations